src/vec.rs

```diff
@@ -844,13 +844,9 @@ impl<T, A: DeallocRef> Vec<T, A> {
     /// Panics if the reallocation fails.
     pub fn into_boxed_slice(self) -> Box<[T], A>
     where
-        A: ReallocRef,
+        A: ReallocRef + Abort,
     {
-        match self.try_into_boxed_slice() {
-            Ok(vec) => vec,
-            Err(CollectionAllocErr::CapacityOverflow) => capacity_overflow(),
-            Err(CollectionAllocErr::AllocError { layout, .. }) => handle_alloc_error(layout.into()),
-        }
+        handle_collection_error_audited(self.try_into_boxed_slice())
```

Comment:
This does not panic, but aborts `CapacityOverflow` now.

Reply:
Really? Scrolling down in my phone (so easy to miss things, take with grain of salt) it looks like I just moved the panic into this `handle_*` function. Did I not?

Reply:
`handle_alloc_error` does not `panic!` but aborts the program

Reply:
Actually it calls a handler, which can me modified, so a capacity overflow has to be handled differently anyways.